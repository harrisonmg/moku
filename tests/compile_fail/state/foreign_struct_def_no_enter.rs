use moku::*;

struct Bottom;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod machine {}

    use machine::BlinkyState;

    struct Top;
    impl TopState<BlinkyState> for Top {}

    use super::Bottom;

    #[superstate(Top)]
    impl State<BlinkyState> for Bottom {}
}

fn main() {}
