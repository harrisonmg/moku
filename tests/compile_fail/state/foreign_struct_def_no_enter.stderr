error: a struct that is not defined in this module must manually implement the `State::enter` function
  --> tests/compile_fail/state/foreign_struct_def_no_enter.rs:20:10
   |
20 |     impl State<BlinkyState> for Bottom {}
   |          ^^^^^
